common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj

```diff
@@ -100,12 +101,28 @@
     (catch ExceptionInfo e
            (handle-es-exception e scroll-id))))
 
+(defn- do-send-with-retry
+  "Sends a query to ES, either normal or using a scroll query."
+  [context index-info query max-retries]
+  (try
+    (if (> max-retries 0)
+      (if-let [scroll-id (:scroll-id query)]
+        (scroll-search context scroll-id)
+        (esd/search
+         (context->conn context) (:index-name index-info) [(:type-name index-info)] query))
+      (errors/throw-service-error :service-unavailable "Exhausted retries to execute ES query"))
+    (catch UnknownHostException e
+      (info (format
+             (str "Excute ES query failed due to UnknownHostException. Retry in half a second."
+                  " Will retry up to %d times.")
+             max-retries))
+      (Thread/sleep 500)
```

Comment:
Might be nice to make this sleep time also configurable

Reply:
Added